src/Workspaces/VisualBasic/Portable/Recommendations/VisualBasicRecommendationService.vb

```diff
@@ -247,6 +247,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Recommendations
             Dim leftHandTypeInfo = context.SemanticModel.GetTypeInfo(leftExpression, cancellationToken)
             Dim leftHandSymbolInfo = context.SemanticModel.GetSymbolInfo(leftExpression, cancellationToken)
 
+            ' GitHub #9087: Try to speculatively bind a type as an expression for My namespace
+            If leftHandTypeInfo.Type IsNot Nothing AndAlso leftHandSymbolInfo.Symbol Is leftHandTypeInfo.Type Then
```

Comment:
> leftHandSymbolInfo.Symbol Is leftHandTypeInfo.Type

Consider using equals to compare symbols, there is no guarantee that both symbols are going to be either reference identical, or not equal.